test/govuk_component/task_list_test.rb

```diff
@@ -87,79 +116,113 @@ def complex_tasklist
   end
 
-  test "renders a complex tasklist" do
-    render_component(groups: complex_tasklist)
+  test "renders links" do
+    render_component(groups: tasklist)
+
+    assert_select group1step1 + " .pub-c-task-list__link-item[href='/link1']", text: "Link 1"
+    assert_select group1step1 + " .pub-c-task-list__link-item[href='/link2']", text: "Link 2"
+    assert_select group1step1 + " .pub-c-task-list__cost", text: "&pound;0 to &pound;300"
+
+    assert_select group2step1 + " .pub-c-task-list__link-item[href='/link3']", text: "Link 3"
+  end
+
+  test "renders links without hrefs" do
+    render_component(groups: tasklist)
+
+    assert_select group2step2 + " .pub-c-task-list__link .pub-c-task-list__link-item[href='/link5']", text: "Link 5"
+    assert_select group2step2 + " .pub-c-task-list__link", text: "or"
+    assert_select group2step2 + " .pub-c-task-list__link .pub-c-task-list__link-item[href='/link6']", text: "Link 6"
+  end
 
-    group1_step1 = group1 + " .pub-c-task-list__step#first-group-first-step"
-    group1_step2 = group1 + " .pub-c-task-list__step#first-group-second-step"
+  test "renders optional steps, sub steps and optional sub steps" do
+    render_component(groups: tasklist)
 
-    assert_select group1 + ".pub-c-task-list__group--active[aria-current='step']"
-    assert_select group2 + ".pub-c-task-list__group--active", false
+    assert_select group1 + " .pub-c-task-list__step.pub-c-task-list__step--optional:nth-of-type(1)"
+    assert_select group1step1 + " .pub-c-task-list__substep .pub-c-task-list__paragraph", text: "This paragraph is inside a required substep"
+    assert_select group2step1 + " .pub-c-task-list__substep.pub-c-task-list__substep--optional .pub-c-task-list__paragraph", text: "This paragraph is inside an optional substep"
+  end
+
+  test "renders get help links back to the main task list" do
+    render_component(groups: tasklist, task_list_url: "/learn-to-drive", task_list_url_link_text: "Get help")
+
+    assert_select group1step1 + " .pub-c-task-list__help-link[href='/learn-to-drive#group-1-step-1']", text: "Get help"
+    assert_select group2step1 + " .pub-c-task-list__help-link[href='/learn-to-drive#group-2-step-1']", text: "Get help"
+  end
+
+  test "group numbering and step logic is displayed correctly" do
+    render_component(groups: tasklist)
+
+    assert_select group1 + " .pub-c-task-list__circle--number .pub-c-task-list__circle-inner .pub-c-task-list__circle-background", text: "Step 1"
+    assert_select group1step2 + " .pub-c-task-list__circle--logic .pub-c-task-list__circle-inner .pub-c-task-list__circle-background", text: "and"
+    assert_select group2 + " .pub-c-task-list__circle--number .pub-c-task-list__circle-inner .pub-c-task-list__circle-background", text: "Step 2"
+    assert_select group2step2 + " .pub-c-task-list__circle--logic .pub-c-task-list__circle-inner .pub-c-task-list__circle-background", text: "or"
+  end
+
+  test "lists have the correct styles" do
```

Comment:
I'd consider renaming this to something more specific about rendering 'required' and 'choice' links differently, as the test isn't really testing "correct styles".

Reply:
Thanks. I'm generally quite bad at thinking of test names.